### ISSUE TITLE
data_stream.py example application is leaking memory

### ISSUE THREAD

**Issue by drogenlied** [enthought/chaco#176]
In 4.4 as well as in 4.3, the data_stream.py example consumes an amount of memory that increases roughly with the amount of points added to the running plot.

This was tested in the Enthought Canopy and python(x,y) distributions under Microsoft Windows 7 32bit.

I'm filing the bug against data_stream.py because it is the most simple publicly available piece of code that displays this behaviour and it is part of Chaco itself, even though the bug and its consequences were discovered in a scientific application that crashed after running for a few days.

**Comment by danieljfarrell** [comment]
@drogenlied did you ever get to the bottom of this. I have an app using Chaco that seems to also be leaking.

**Comment by drogenlied** [comment]
@danieljfarrell no, we modernized our application, now the stack is Python3/PyQt5/pyqtgraph
